Добавлены тесты, локаторы, фикстуры, методы отдельно

grayyyyy456 opened the pull request.

grayyyyy456 committed:
Добавлены тесты, локаторы, фикстуры, методы отдельно

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,11 @@
+@pytest.fixture(scope="function")
+def browser():
+    driver = webdriver.Firefox()
+    driver.get("https://qa-scooter.praktikum-services.ru/")
```
Можно лучше: здесь и далее: урлы стоит хранить в отдельном модуле и конкатенировать с базовым, так будет проще поддерживать несколько стендов

ShamilovR commented on pages/home_page.py:
```diff
@@ -0,0 +1,59 @@
+from locators.home_page_locators import Locators
+
+
+class HomePageMesto:
```
Нужно исправить: необходимо реализовать базовый класс и вынести туда все методы для работы с драйвером и элементами. Далее пейджи должны наследоваться от базового класса и переиспользовать его методы. Базовый класс - абстракция, созданная только для наследования.

ShamilovR commented on pages/home_page.py:
```diff
@@ -0,0 +1,59 @@
+        self.driver = driver
+    
+    def wait_for_load_questions(self):
+        WebDriverWait(self.driver, 5).until(expected_conditions.visibility_of_element_located(Locators.questions))
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR commented on pages/home_page.py:
```diff
@@ -0,0 +1,59 @@
+    def __init__(self, driver):
+        self.driver = driver
+    
+    def wait_for_load_questions(self):
```
Нужно исправить: здесь и далее: отсутствует декоратор step над методом. Его необходимо прописывать над методами пейджей для информативного allure отчета.

ShamilovR commented on pages/order_1_page.py:
```diff
@@ -0,0 +1,47 @@
+import time
```
Нужно исправить: неиспользуемый импорт модуля.

ShamilovR commented on pages/order_2_page.py:
```diff
@@ -0,0 +1,45 @@
+    def click_order_button(self):
+        WebDriverWait(self.driver, 10).until(expected_conditions.visibility_of_element_located(Locators.order_button))
+        self.driver.find_element(*Locators.order_button).click()
+        time.sleep(2)
```
Нужно исправить: sleep в тестах плохая практика, т.к. для загрузки страницы/элемента может потребоваться меньше или больше времени. Для ожиданий нужно использовать модуль expected_condition.

ShamilovR commented on tests/test_click_order_buttons.py:
```diff
@@ -0,0 +1,27 @@
+from selenium.webdriver.support import expected_conditions
+from locators.home_page_locators import Locators
+from pages.home_page import HomePageMesto
+from conftest import browser
```
Нужно исправить: здесь и далее: фикстуры не нужно импортировать. Если они лежат в conftest то pytest их видит без импорта

ShamilovR commented on tests/test_click_order_buttons.py:
```diff
@@ -0,0 +1,27 @@
+
+class TestClickOrderButtons:
+
+    def test_click_1_order_button(self, browser):
```
Нужно исправить: здесь и далее: для читаемого отображения отчетов над каждым тестовым методом необходимо ставить декоратор title.

ShamilovR commented on tests/test_click_order_buttons.py:
```diff
@@ -0,0 +1,27 @@
+    def test_click_1_order_button(self, browser):
+        first_button = HomePageMesto(browser)
+        first_button.wait_for_load_questions()
+        first_button_click = browser.find_element(*Locators.order_button_up)
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR replied:
Нужно исправить: здесь и далее: в классическом POM тест не должен знать о внутренней реализации, в том числе и о используемых локаторах (за исключением передачи локаторов в рамках параметризации). Подобная реализация должна быть сокрыта в рамках методов пейджи.

ShamilovR commented on tests/test_fill_out_order_form.py:
```diff
@@ -0,0 +1,32 @@
+class TestFillOutOrderForm:
+
+    @pytest.mark.parametrize("name, last_name, address, metro, phone, comment", [
+        ("Иван", "Иванов", "Усачева 3", "Сокольники", "12312312311", "zakaz1"),
```
Можно лучше: здесь и далее: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

ShamilovR commented on tests/test_questions_click.py:
```diff
@@ -0,0 +1,87 @@
+
+class TestClickQuestions:
+
+    def test_question_1_expands_correctly(self,browser):
```
Можно лучше: для данных тестовых сценариев лучше использовать параметризацию

grayyyyy456 committed:
Все url перенесены в отдельный файл

grayyyyy456 committed:
Создан базовый класс, туда вынесены методы дял работы с драйвером и элементами
Так же теперь в home_page.py не используются обращение к драйверу и WebdriverWait

grayyyyy456 committed:
Добавлены декораторы step для всех методов в файле home_page.py

grayyyyy456 committed:
Убран неиспользуемый импорт модуля в файле order_1_page.py
Убран sleep в файле order_2_page.py

grayyyyy456 committed:
Убран импорт фикстуры во всех тестах

grayyyyy456 committed:
Исправлены другие pages

grayyyyy456 committed:
Над каждым тестом поставлен декоратор title

grayyyyy456 committed:
убраны ненужные импорты

grayyyyy456 committed:
исправлены тесты и использование методов

grayyyyy456 committed:
добавлены step в base_page.py

ShamilovR commented on pages/home_page.py:
```diff
@@ -0,0 +1,66 @@
+
+    @allure.step("Переключение на последнюю открытую вкладку")
+    def switch_to_last_tab(self):
+        self.driver.switch_to.window(self.driver.window_handles[-1])
```
Нужно исправить: здесь и далее: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR commented on pages/order_1_page.py:
```diff
@@ -0,0 +1,54 @@
+from pages.base_page import BasePage
+from pages.home_page import HomePageMesto
+
+class OrderPage1Mesto(HomePageMesto, BasePage):
```
Можно лучше: HomePageMesto наследник BasePage. Если класс наследуется от HomePageMesto, то от BasePage наследоваться не нужно, т.к. OrderPage1Mesto будет наследником BasePage через HomePageMesto

ShamilovR commented on tests/test_yandex_button_in_logo.py:
```diff
@@ -0,0 +1,18 @@
+        logo_yandex_button.wait_for_load_questions()
+        logo_yandex_button.click_yandex_button()
+        logo_yandex_button.switch_to_last_tab()
+        WebDriverWait(browser, 10).until(expected_conditions.url_to_be(url_yandex))
```
Нужно исправить: в тестах и методах пейджей не должно быть обращений к драйверу и использований WebdriverWait. Все подобные реализации должны находиться в рамках базового класса пейджей.

ShamilovR commented on tests/test_click_order_buttons.py:
```diff
@@ -0,0 +1,24 @@
+    def test_click_1_order_button(self, browser):
+        first_button = HomePageMesto(browser)
+        first_button.wait_for_load_questions()
+        first_button.click_element(Locators.order_button_up)
```
Нужно исправить: здесь и далее: в классическом POM тест не должен знать о внутренней реализации, в том числе и о используемых локаторах (за исключением передачи локаторов в рамках параметризации). Подобная реализация должна быть сокрыта в рамках методов пейджи.

grayyyyy456 committed:
В пейджах кроме base_page.py убраны обращение к driver и WebdriverWait. В тестах нет прямого обращение к локаторам

grayyyyy456 committed:
last commit
